modules/renter/download.go

```diff
@@ -63,20 +65,23 @@ type (
 		atomicDataReceived uint64
 		downloadComplete   bool
 		downloadErr        error
-		finishedChunks     []bool
+		finishedChunks     map[int]bool
```

Comment:
This map seems really memory intensive to me, especially given that progress should be approximately linear. I'm probably worrying too much about memory footprint at this point though.